src/main/java/org/entando/plugins/pda/pam/service/process/KieProcessService.java

```diff
@@ -0,0 +1,66 @@
+package org.entando.plugins.pda.pam.service.process;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.entando.plugins.pda.core.engine.Connection;
+import org.entando.plugins.pda.core.model.ProcessDefinition;
+import org.entando.plugins.pda.core.service.task.ProcessService;
+import org.entando.plugins.pda.pam.service.KieUtils;
+import org.entando.plugins.pda.pam.service.process.model.KieProcessDefinition;
+import org.entando.plugins.pda.pam.service.task.model.KieProcessDefinitionsResponse;
+import org.entando.web.exception.BadResponseException;
+import org.entando.web.request.PagedListRequest;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class KieProcessService implements ProcessService {
+
+    private final RestTemplateBuilder restTemplateBuilder;
+
+    //CHECKSTYLE:OFF
+    public static final String PROCESS_DEFINITION_LIST_URL = "/queries/processes/definitions";
+    //CHECKSTYLE:ON
+
+    @Override
+    public List<ProcessDefinition> listDefinitions(Connection connection) {
+        RestTemplate restTemplate = restTemplateBuilder
+                .basicAuthorization(connection.getUsername(), connection.getPassword())
+                .build();
+
+        List<ProcessDefinition> result = new ArrayList<>();
+
+        PagedListRequest pageRequest = new PagedListRequest();
+        pageRequest.setPageSize(2_000_000_000); //Set max page size to get all results
```

Comment:
Could you create a constant for this number to improve readability? MAX_PAGE_SIZE maybe?